### ISSUE TITLE
RC 8 fails with "Failed to run command 'listdatabases'" 

### ISSUE THREAD

**Issue by jeffwidman** [paralect/robomongo#1109]
I have a SSH tunnel setup in my `~/.ssh/config` file that maps a local port to a remote Mongo server. 

Accessing the Mongo database worked fine on RC 7. 

I just upgraded to RC 8 and got the error "Failed to run command 'listdatabases'" 

I reverted to RC 7 and everything works fine again.

**Comment by simsekgokhan** [comment]
Hi Jeff, thanks for reporting the issue. We will try to reproduce and investigate the issue. Can you also give more information about OS versions of your local, jumphost and mongo server?

**Comment by jeffwidman** [comment]
Local: OS El Cap (latest Stable release)
Jumphost: Ubuntu 14.04
Mongo: Ubuntu 14.04

Since the tunnel uses `ProxyCommand` through the jumphost, the jumphost should have zero impact...

**Comment by psypersky** [comment]
Hello, I have this error too, I created a mongodb on docker exposed the port to the host and then tried to connect from my host to the port

Local Host: OS X 10.13.6
Docker version: 18.06.1-ce-mac73
Docker container: mongo:4.1.2-xenial

It looks like it is actually connecting, here are my mongo logs

```
2018-09-02T15:39:09.356+0000 I NETWORK  [listener] connection accepted from 172.17.0.1:40922 #4 (1 connection now open)
2018-09-02T15:39:09.358+0000 I NETWORK  [conn4] received client metadata from 172.17.0.1:40922 conn4: { application: { name: "robo3t" }, driver: { name: "MongoDB Internal Client", version: "3.4.3-10-g865d2fb" }, os: { type: "Darwin", name: "Mac OS X", architecture: "x86_64", version: "17.7.0" } }
2018-09-02T15:39:09.360+0000 E -        [conn4] Assertion: Location34348: cannot translate opcode 2010 src/mongo/rpc/message.h 119
2018-09-02T15:39:09.361+0000 I NETWORK  [conn4] DBException handling request, closing client connection: Location34348: cannot translate opcode 2010
2018-09-02T15:39:09.361+0000 I NETWORK  [conn4] end connection 172.17.0.1:40922 (0 connections now open)
```

**Comment by pierbettassa** [comment]
> 
> 
> Hello, I have this error too, I created a mongodb on docker exposed the port to the host and then tried to connect from my host to the port
> 
> Local Host: OS X 10.13.6
> Docker version: 18.06.1-ce-mac73
> Docker container: mongo:4.1.2-xenial
> 
> It looks like it is actually connecting, here are my mongo logs
> 
> ```
> 2018-09-02T15:39:09.356+0000 I NETWORK  [listener] connection accepted from 172.17.0.1:40922 #4 (1 connection now open)
> 2018-09-02T15:39:09.358+0000 I NETWORK  [conn4] received client metadata from 172.17.0.1:40922 conn4: { application: { name: "robo3t" }, driver: { name: "MongoDB Internal Client", version: "3.4.3-10-g865d2fb" }, os: { type: "Darwin", name: "Mac OS X", architecture: "x86_64", version: "17.7.0" } }
> 2018-09-02T15:39:09.360+0000 E -        [conn4] Assertion: Location34348: cannot translate opcode 2010 src/mongo/rpc/message.h 119
> 2018-09-02T15:39:09.361+0000 I NETWORK  [conn4] DBException handling request, closing client connection: Location34348: cannot translate opcode 2010
> 2018-09-02T15:39:09.361+0000 I NETWORK  [conn4] end connection 172.17.0.1:40922 (0 connections now open)
> ```

Same here:
Local Host: Windows 10
Docker version: 18.09.0 running on Ubuntu 16.04.5LTS - kernel 4.15.0-38-generic
Docker container: bitnami/mongodb:4.1.4-debian-9-r2

```
2018-11-13T10:31:08.317+0000 I NETWORK  [listener] connection accepted from 192.168.1.247:49839 #24 (11 connections now open)
2018-11-13T10:31:08.318+0000 I NETWORK  [conn24] received client metadata from 192.168.1.247:49839 conn24: { application: { name: "robo3t" }, driver: { name: "MongoDB Internal Client", version: "3.4.3-10-g865d2fb" }, os: { type: "Windows", name: "Microsoft Windows 8", architecture: "x86_64", version: "6.2 (build 9200)" } }
2018-11-13T10:31:08.321+0000 E -        [conn24] Assertion: Location34348: cannot translate opcode 2010 src/mongo/rpc/message.h 119
2018-11-13T10:31:08.321+0000 I NETWORK  [conn24] DBException handling request, closing client connection: Location34348: cannot translate opcode 2010
2018-11-13T10:31:08.321+0000 I NETWORK  [conn24] end connection 192.168.1.247:49839 (10 connections now open)
```

**Comment by vrst37** [comment]
+1

**Comment by diogoffmelo** [comment]
+2

**Comment by BernMcCarty** [comment]
+3

**Comment by vrst37** [comment]
I do not think this issue should be closed. This issue occurs when an invalid username or password is provided for connection. 
The error message must be changed to something non cryptic (like invalid credentials or something). 

What do you guys think?

**Comment by vrst37** [comment]
I do not think this issue should be closed. This issue occurs when an invalid username or password is provided for connection. 
The error message must be changed to something non cryptic (like invalid credentials or something). 

What do you guys think?

**Comment by syzer** [comment]
+1

**Comment by adityathebe** [comment]
> I do not think this issue should be closed. This issue occurs when an invalid username or password is provided for connection.
> The error message must be changed to something non cryptic (like invalid credentials or something).
> 
> What do you guys think?

I don't have any authentication set up but I still get this error. I don't think it's due to invalid username or password.

**Comment by nemasobhani** [comment]
+1
